### DISCUSSION TITLE
Combobox option to autoexpand list on focus

### DISCUSSION THREAD

**Discussion by mattmutt** [vmware/clarity#6131] [Ideas]
An optional feature consideration the combobox.

When the user brings focus to the input field of the combobox , invoke the `ClrPopoverToggleService` to expand the list instead of making the user move mouse to expand trigger button on the right side or press the down arrow key to reveal the list.

Currently the list would show one item if some text were already entered, or the complete list if nothing was in the input text. I'm not sure if showing the entire list might ever be useful, to saving the user from having to clear out the input prior to seeing a complete list, like a select box.

https://user-images.githubusercontent.com/8688570/124859546-fe3bfd00-df64-11eb-8241-ace66584e00d.mov

**Comment by Jinnie** [comment]
What you suggest, @mattmutt, was the original behavior we had implemented. But it turned out not to be aligned with the A11Y requirements and W3C examples. Adding a configuration option would be a viable approach, but we don't want overload our API and we're trying to avoid introducing API changes in Clarity Angular.
This is a good case to illustrate the advantages of the Clarity Core approach, where such varying user requirements will be much easier to implement on the user side.

**Reply by d-m-s** [reply]
Autoexpand is one issue, but is it really helpful to just see the current selected value once again in the dropdown-suggestion-list?

Of course when the whole text in the input field is highlighted the user can quickly type something in, but he/she would still need to know which other valid options exist by heart.

So probably it would be more helpful once the combobox is populated to show all valid options in the suggestion-list (as also suggested in #6469)
